guidance/src/guidance/guidance_worker.cpp

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright (C) 2018-2019 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+#include "guidance/guidance_worker.hpp"
+
+namespace guidance
+{
+    GuidanceWorker::GuidanceWorker() 
+    {
+        nh_ = ros::CARMANodeHandle{};
+        pnh_ = ros::CARMANodeHandle{"~"};
+    }
+
+    void GuidanceWorker::populate_plugin_list_response(cav_srvs::PluginListResponse& res)
+    {
+        for(int i = i; i < plugins.size(); ++i)
+        {
+            res.plugins.push_back(plugins[i]);
+        }
+    }
+
+    void GuidanceWorker::populate_active_plugin_list_response(cav_srvs::PluginListResponse& res)
+    {
+        for(int i = i; i < plugins.size(); ++i)
+        {
+            if(plugins[i].activated)
+            {
+                res.plugins.push_back(plugins[i]);
+            }
+        }
+    }
+
+    bool GuidanceWorker::is_required_plugin(std::string plugin_name, std::string version)
+    {
+        for(int i = 0; i < required_plugins.size(); i++)
+        {
+            if(plugin_name.append(" " + version).compare(required_plugins[i]) == 0)
```

Comment:
Rather than adding a single space with the version number it would be more robust to strip spaces from the required_plugin list at startup and compare without spaces. Then if someone accidentally types "Autoware Plugin   1.0.0" (extra spaces). It will not fail to find a match

Reply:
Sounds good to me.